### ISSUE TITLE
dgl.prop_nodes_topo Message Passing Error

### ISSUE THREAD

**Issue by bkong999** [dmlc/dgl#2484]
## 🐛 Bug

Clearly, there is something wrong with the message passing in the following example, either because source nodes' features are passed to the wrong dst nodes or the final updated features are written to different nodes.

Basically, I use this function to aggregate the non-negative predictions (or sum the upstreaming features including the current node's feature) from the root to the leaves in a tree. As the graph is tree-structured and the predictions are non-negative, the aggregated value should monotonically increase from the root to the leaves. The following shows the result of an 8-path tree using DGL v0.4.3. x-axis means the indices in each path. y-axis is the aggregated value.
![v4](https://user-images.githubusercontent.com/61814189/103498209-ca86eb80-4df8-11eb-8a4b-5cae38ac4860.png)

However, when I switched from v0.4.3 to v0.5.3, the aggregated values are not monotonically increasing anymore. The following shows the result using DGL v0.5.3. Apparently, there is something wrong with this function. Tried both the built-in message passing functions and writing by myself, the results are inconsistent for these versions.
![v5](https://user-images.githubusercontent.com/61814189/103498273-002bd480-4df9-11eb-8573-a8ad60098dd6.png)

## To Reproduce

1. Create a large tree-structured graph (the nodes have either one or zero in degrees).
2. Write non-negative feature into the node data.
3. Propagate the messages among the graph as follows:
```
dgl.prop_nodes_topo(
      graph,
      message_func=dgl.function.u_add_v('data', 'data', 'data'),
      reduce_func=dgl.function.sum('data', 'data'))
```

The synthetic data and script to reproduce this issue can be found in the attached zip file.

[reproduce.zip](https://github.com/dmlc/dgl/files/5762879/reproduce.zip)

## Expected behavior

the aggregated values should be monotonically increasing from the root to the leaves.
## Environment

 - DGL Version: 0.5.3
 - Backend Library & Version: PyTorch 1.7.0
 - OS (Linux): Linux
 - How you installed DGL (`conda`): conda install -c dglteam dgl-cuda10.1
 - Python version: 3.7.4
 - CUDA/cuDNN version (if applicable): 10.1
 - GPU models and configuration (e.g. V100): Tesla P40
 - Any other relevant information: The synthetic data and script to reproduce this issue can be found in the zip file.

**Comment by BarclayII** [comment]
Seems to work for me.  Although I'm not sure if it matches your case:
```python
import dgl
import torch
g = dgl.graph(([0,0,1,1,2,2,3,3,4,4,5,5,6,6],[1,2,3,4,5,6,7,8,9,10,11,12,13,14]))
x = torch.rand(15)
g.ndata['x'] = x
dgl.prop_nodes_topo(g, dgl.function.u_add_v('x', 'x', 'm'), dgl.function.sum('m', 'x'))
print(g.ndata['x'])
for i in range(1, 15):
    assert g.ndata['x'][i] > g.ndata['x'][(i - 1) // 2]
```

Could you give an example code so that I can reproduce your problem?  Thanks.

**Comment by bkong999** [comment]
I guess you need a large enough tree. I have already attached the data and
the script to reproduce it in the issue:
https://github.com/dmlc/dgl/files/5762879/reproduce.zip

Best

**Comment by BarclayII** [comment]
Yeah I noticed that.  Bug confirmed.  Seems that there is a mistake in message passing graph generation.

**Comment by BarclayII** [comment]
Should be fixed and ready in the next nightly build.  Please feel free to reopen if it does not work for you.
